core/src/test/java/org/apache/calcite/test/RelOptRulesTest.java

```diff
@@ -6448,6 +6448,42 @@ private void checkSemiJoinRuleOnAntiJoin(RelOptRule rule) {
         .check();
   }
 
+  /** Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-5680">[CALCITE-5680]
+   * Wrong field reference lookup due to same intermediate table alias
+   * of multiple sub-queries with only literal operands in subquery remove phase</a>. */
+  @Test void testExpandFilterConstantInCorrelatedWithTwoSubQueries() {
+    final String sql = "select empno from sales.empnullables as e\n"
+        + "where 1 in (\n"
+        + "  select deptno from sales.deptnullables where e.ename = name and deptno > 10)\n"
+        + "or 2 in (\n"
+        + "  select deptno from sales.deptnullables where e.ename = name and deptno < 20)";
+    System.out.println(sql);
+    sql(sql)
+        .withSubQueryRules()
+        .withRelBuilderSimplify(false)
+        .withTrim(true)
+        .check();
+  }
+
+  /** Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-5680">[CALCITE-5680]
+   * Wrong field reference lookup due to same intermediate table alias
+   * of multiple sub-queries with only literal operands in subquery remove phase</a>. */
+  @Test void testExpandFilterConstantInWithTwoSubQueries() {
+    final String sql = "select empno from sales.empnullables\n"
+        + "where 1 in (\n"
+        + "  select deptno from sales.deptnullables where name = 'dept1')\n"
+        + "or 2 in (\n"
+        + "  select deptno from sales.deptnullables where name = 'dept2')";
+    System.out.println(sql);
+    sql(sql)
+        .withSubQueryRules()
+        .withRelBuilderSimplify(false)
```

Comment:
Is the plan correct  if we enable RelBuilderSimplify?

Reply:
After this fix, yes. We here don't enable RelBuilderSimplify, to make plan more original, and easier to identify that the plan is correct.

Reply:
Ok, I understand.

I suggest you write a note to explain it, otherwise others may mistakenly think that this is a reproduction condition.

Reply:
Thanks for your suggestion, I will add it soon.

Reply:
Added.